components/resources/library/src/webMain/kotlin/org/jetbrains/compose/resources/Resource.web.kt

```diff
@@ -50,4 +57,111 @@ internal fun getResourceUrl(windowOrigin: String, windowPathname: String, resour
         path.startsWith("http://") || path.startsWith("https://") -> path
         else -> windowOrigin + windowPathname + path
     }
+}
+
+/**
+ * Preloads a font resource and provides a [State] containing the loaded [Font] or `null` if not yet loaded.
+ *
+ * Internally, it reads font bytes, converts them to a [Font] object, and caches the result, speeding up future
+ * accesses to the same font resource when using @Composable Font function.
+ *
+ * **Usage Example:**
+ * ```
+ * @Composable
+ * fun MyApp() {
+ *     val fontState by preloadFont(Res.font.SomeNiceFont)
+ *
+ *     if (fontState != null) {
+ *         Text(
+ *             text = "Hello, World!",
+ *             fontFamily = FontFamily(Font(Res.font.SomeNiceFont)) // the font is taken from the cache
+ *         )
+ *     }
```

Comment:
Let's add an else branch with a Progress as well here?

Reply:
done in all examples